community/louvain_common.go

```diff
@@ -35,11 +36,194 @@ func Q(g graph.Graph, communities [][]graph.Node, resolution float64) float64 {
 	}
 }
 
+// ReducedGraph is a modularised graph.
+type ReducedGraph interface {
+	graph.Graph
+
+	// Communities returns the community memberships
+	// of the nodes in the graph used to generate
+	// the reduced graph.
+	Communities() [][]graph.Node
+
+	// Structure returns the community structure of
+	// the current level of the module clustering.
+	// The first index of the returned value
```

Comment:
Structure returns the reduction that was applied to the result of the previous iteration of the Louvain algorithm to get the reduced graph represented by the receiver. I'd be happy if you can figure out a nice way of saying that.